src/Citeproc/Eval.hs

```diff
@@ -73,6 +73,10 @@ data EvalState a =
   , stateRefMap         :: ReferenceMap a
   , stateReference      :: Reference a
   , stateUsedYearSuffix :: Bool
+  , stateUsedIdentifier :: Bool
+  -- ^ tracks whether an identifier (DOI,PMCID,PMID,URL) has yet been used
+  , stateUsedTitle      :: Bool
+  -- ^ tracks whether the item title has yet been used
```

Comment:
Do we really need these?
We could traverse to see if we have a Linked or a TagTitle.
If we have a Linked, then it means we had one of these identifiers and we're done.
If no Linked, but TagTitle, then we can put a Linked on the title.
If none of the above, put a Linked on the whole thing.

Reply:
I suppose it's slightly more efficient to track this in state, so maybe that justifies this.